### PR TITLE
[ARM] `az deployment group create`: Make `--template-file` parameter optional when used with `.bicepparam` parameter file

### PR DESCRIPTION
**Pull request by anthony-c-martin**
**Related command**

`az deployment`

**Description**

In order to support `.bicepparam` files which link directly to Bicep Registry, Template Spec or JSON content, we must relax the requirement for the user to provide the `--template-file` parameter for deployment-related commands. This parameter is currently redundant, as `.bicepparam` files require the user to link to a `.bicep` file via the `using` statement.

To maintain backwards-compatibility, continue to support the "old" command if the user supplies the `--template-file` parameter.

**Testing Guide**
```sh
az deployment group create \ 
  -g myRg \
  -p .\bicepparam\storage_account_params.bicepparam  
```

**Notes**
- Integration tests in this PR are `LiveScenarioTest` because we need to do a Bicep CLI version check (meaning it has to be installed) before performing any `.bicepparam` related logic
- Corresponding change in AzPwsh has been merged: https://github.com/Azure/azure-powershell/pull/22728

**History Notes**

[ARM] `az deployment group create`: Make `--template-file` parameter optional when used with `.bicepparam` parameter file

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by anthony-c-martin** [comment]
@zhoxing-ms would it be possible to get this reviewed please? The Azure PowerShell counterpart PR has been merged (https://github.com/Azure/azure-powershell/pull/22728)

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -1172,24 +1187,9 @@ def _prepare_stacks_templates_and_parameters(cmd, rcf, deployment_stack_model, t
...
-                raise ArgumentUsageError("Only a .bicep template is allowed with a .bicepparam parameter file")
-
-            build_bicepparam_output = run_bicep_command(
-                cmd.cli_ctx, ["build-params", bicepparam_file, "--bicep-file", template_file, "--stdout"])
-            build_bicepparam_output_json = json.loads(build_bicepparam_output)
-            template_content = build_bicepparam_output_json["templateJson"]
-            bicepparam_json_content = build_bicepparam_output_json["parametersJson"]
+            template_content, template_spec_id, bicepparam_json_content = _parse_bicepparam_file(cmd.cli_ctx, template_file, parameters)
+            if template_spec_id:
+                deployment_stacks_template_link = rcf.deployment_stacks.models.DeploymentStacksTemplateLink(id=template_spec_id)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py:1192]: Please use `cmd.get_models()` to get the SDK model

**Comment by anthony-c-martin** [comment]
@zhoxing-ms please could you take another look?

**Review by zhoxing-ms — APPROVED** [review]

**Review by shenglol — APPROVED** [review]
:shipit:

**Pull request merged**